### ISSUE TITLE
Feature Request: Per Device Permissions

### ISSUE THREAD

**Issue by EULABreaker** [Ylianst/MeshCentral#164]
Add the same permissions from the Device Groups to the Machine level.

It would be great to be able to allow user to remote to one machine in a group without giving access to the whole group. With this you could allow a user access to there own machine but still keep everything organized in groups.

Example:
I have device group "Desktops" with multiple machines in it. I want to allow Bill to connect to his machine. Currently I would have to give him permissions to the group so he could remote to his machine. But this would also allow him to remote to other user machines in the group. If I could choose the permission on the Machine level it would allow Bill to access his machine but no others in that group.

**Comment by merlinweb** [comment]
Great improvement…

Downside, this will open MC2 to a number of end user…

**Comment by D4V3M0NK** [comment]
An alternative would be to have the ability to attribute (one or more) machines to a user account? If it were done that way, Bill wouldn't see any device groups at all, let alone the machines within those groups: just the number of entries that he had access to, within **My Devices**.

**Comment by EULABreaker** [comment]
That does sound like a better and more sound approach to the issue.

**Comment by corriganc** [comment]
Do we have any updates on this? I too am interested in assigning devices to specific users or at least preventing them from accessing particular machines.

We're hoping to use MeshCentral as a remote access solution for our staff, but we don't want, for example, Bob from Design to be able to access Sarah from Accounting's PC due to the sensitive information that may be on the screen/available through the file viewer.

If this could be implemented, I think MeshCentral would definitely be rolled out as our Remote Access solution, replacing Citrix XenDesktop/XenApp.

**Comment by GusevVictor** [comment]
I vote for this problem too! Please add this possibility

**Comment by Ylianst** [comment]
I changed the MeshCentral access-control system to allow user groups a few weeks back, the good news is that I think that with the new system, I should be able to add per-device permissions. I am traveling all this week, but this is certainly one thing I want to look into.

**Comment by SomeGuru** [comment]
I seen this enhancement in the latest servers, would this then close this issue if we validate?

**Comment by JSuenram** [comment]
@EULABreaker 
I believe this is already fixed a while ago.
Please close if so.

**Comment by si458** [comment]
@JSuenram you are correct, this has been added in the latest versions so ill close this issue
